python/cuml/test/test_metrics.py

```diff
@@ -1103,6 +1105,173 @@ def test_pairwise_distances_output_types(input_type, output_type, use_global):
             assert isinstance(S, cp.core.core.ndarray)
 
 
+@pytest.mark.parametrize("metric", PAIRWISE_DISTANCE_SPARSE_METRICS.keys())
+@pytest.mark.parametrize("matrix_size", [(3, 8), (800, 10), (2, 400),
+                                         (500, 40)])
+def test_sparse_pairwise_distances(metric: str, matrix_size):
+    # Test the sparse_pairwise_distance helper function.
+    if metric in ["inner_product", "hellinger"]:
+        pytest.skip('Skipping metrics not supported by sklearn')
+    if not has_scipy():
+        pytest.skip('Skipping sparse_pairwise_distances if Scipy is missing')
+    import scipy
+
+    # For fp64, compare at 13 decimals, (2 places less than the ~15 max)
+    compare_precision = 7
+
+    # Compare to sklearn, single input
+    X = scipy.sparse.random(matrix_size[0], matrix_size[1],
+                            dtype=np.float64, random_state=123, density=0.6)
```

Comment:
we should parametrize density. 0.6 I don't think we'd see that in practice. Perhaps we could even parametrize the density w/ the matrix size so we could create a larger matrix with lower density to verify. This also makes it easier to plug in different values when debugging / profiling.

Reply:
Yes, this high density was just to have numbers to compare instead of just zeros. I parametrized it.